### ISSUE TITLE
Scope improvements to Schedule App

### ISSUE THREAD

**Issue by edrushka** [MozillaFoundation/plan#710]
During this milestone, I'll host a meeting with @davidascher @simonwex @ryanpitts and @xmatthewx to discuss future improvements to the Schedule App. Hopefully we can talk about timelines wrt dev capacity and upcoming events (SRCCON is likely the next event to need the app).

Let me know if you're interested in attending this meeting and are not listed above.

**Comment by davidascher** [comment]
I'm in India during this heartbeat, so that meeting may be tough depending on the time of day.  I'm on PTO second week, so next week may be possible.

Missing some context to this ticket, but:

* I think @adamlofting would be useful, as IMO the schedule app has the potential of being useful both as a probe of the effect of our convenings, as well as a tool to measure sub-indicators towards network health.

* I'd love for us to build a roadmap for the schedule app, even if aspirational, that we can both use to assess a long-term technical plan, but also to assess what it can do for participants, organizers, and Mozilla.  Feels like a discussion that will take a while, and that should include event leaders and community managers.

TLDR: We should understand the tech implications (status + possibilities), but I'm more thirsty for a vision that maps the activities of community member, hubs, the realities of the events themselves and the impact we think the app can have.  I think that vision will likely span many heartbeats -- that's ok, as long as we think it can yield significant value to the overall network health (beyond just displaying schedule information).

**Comment by simonwex** [comment]
@mmmavis is interested in continuing to work on this project, could you please include her, too?

**Comment by cassiemc** [comment]
I'm interested in this as well @edrushka

**Comment by xmatthewx** [comment]
I'm curious -- were any advances made with the Chicago iteration? (I know it was a very tight timeline.) Any progress on separating the "theme" from the core app?

**Comment by edrushka** [comment]
@ScottDowne and @mmmavis can you weigh in? Mavis, please explain your last-minute gdoc --> json hack, even though I'm not actually sure that we used it in the end.

**Comment by ScottDowne** [comment]
@xmatthewx @edrushka I moved it into a single repo. Nothing flashy.

It wasn't as simple as I hoped, so I ended up having to cut a bunch of corners to ship the app in time for Chicago. Needs another pass.

**Comment by adamlofting** [comment]
I'd love to learn more (and attend the call).

**Comment by mmmavis** [comment]
I didn't have the confidence to do to much rewrite within the tight timeframe.  To minimize changes to the existing app all I did was formatting the exported the data from Google Spreadsheet to its MozFest Schedule equivalent (e.g, Moonshot -> Pathway etc). 

We didn't have the time to modify the Python script nor set up any cron job for fetching and pushing Spreadsheet changes. As an quick alternative, I got [this data exporter thing](http://mmmavis.github.io/hive-chicago-buzz-data-downloader/) up last min for @edrushka to export Spreadsheet data in the format the Scheduler takes.  Basically anyone can just hit the "Download JSON" button and update the `session.json` in the repo (using Github's interface) with that .json file downloaded from that exporter thing.

**Comment by simonwex** [comment]
@edrushka did this call happen?

**Comment by xmatthewx** [comment]
Meeting will be rescheduled due to conflict for several key people.

**Comment by edrushka** [comment]
We had a good meeting about this last week. Here's the MVP Proposal:

**1. Create a single App that can be themed for a specific event**
Problem: We have mixed event customization with app iteration. Improvents in Chicago instance might require work to bring them into SRCCON or Mozfest apps. 
Solution: Move event-specific bits (e.g. branding, help link, turn on/off "spaces", highlight colour) to a settings/theme folder where we can adjust them for each event without creating an entirely new instance.

**2. Enable non-devs to edit and push changes right to last minute**
Problem: Event organizers need a surefire way to update the schedule information on the fly without pinging a dev. 
Solution: Ensure there is both a Python + Node version of the data loader so we don’t become dependent on dev-heavy workarounds. Expand things that can set by non-devs (categories, locations etc.)

In addition, @xmatthewx will host a design prototyping heartbeat to explore the myriad nice-to-have feature ideas that have been proposed.

Closing this scoping ticket and will file more tickets for individual pieces of work.
